src/server/server.ts

```diff
@@ -1,127 +1,38 @@
-import http from 'http'
-
-export async function start (options: ListenOptions): Promise<void> {
-  let _remoteExtensionHostAgentServer: IServerAPI | null = null
-  let _remoteExtensionHostAgentServerPromise: Promise<IServerAPI> | null = null
-  const getRemoteExtensionHostAgentServer = (): Promise<IServerAPI> => {
-    if (_remoteExtensionHostAgentServerPromise == null) {
-      _remoteExtensionHostAgentServerPromise = createServer(address).then(server => {
-        _remoteExtensionHostAgentServer = server
-        return server
-      })
-    }
-    return _remoteExtensionHostAgentServerPromise
-  }
-
-  let address: AddressInfo | string | null = null
-  const server = http.createServer(async (req, res) => {
-    const remoteExtensionHostAgentServer = await getRemoteExtensionHostAgentServer()
-    return remoteExtensionHostAgentServer.handleRequest(req, res)
-  })
-  server.on('upgrade', async (req, socket) => {
-    const remoteExtensionHostAgentServer = await getRemoteExtensionHostAgentServer()
-    return remoteExtensionHostAgentServer.handleUpgrade(req, <Socket>socket)
-  })
-  server.on('error', async (err) => {
-    const remoteExtensionHostAgentServer = await getRemoteExtensionHostAgentServer()
-    return remoteExtensionHostAgentServer.handleServerError(err)
-  })
-
-  server.listen(options, async () => {
-    address = server.address()
-    if (address === null) {
-      throw new Error('Unexpected server address')
-    }
-
-    // eslint-disable-next-line no-console
-    console.log(`Server bound to ${typeof address === 'string' ? address : `${address.address}:${address.port} (${address.family})`}`)
-
-    await getRemoteExtensionHostAgentServer()
-  })
-
-  process.on('exit', () => {
-    server.close()
-    if (_remoteExtensionHostAgentServer != null) {
-      _remoteExtensionHostAgentServer.dispose()
-    }
-  })
+import { IProductService } from 'vs/platform/product/common/productService'
+import * as fs from 'fs/promises'
+import * as path from 'path'
+import { fileURLToPath } from 'url'
+
+const thisWithVSCodeParams = globalThis as typeof globalThis & {
+  _VSCODE_PRODUCT_JSON: Partial<IProductService>
+  _VSCODE_PACKAGE_JSON: { version: string }
 }
 
-function parseRange (strRange: string) {
-  const match = strRange.match(/^(\d+)-(\d+)$/)
-  if (match != null) {
-    const start = parseInt(match[1]!, 10); const end = parseInt(match[2]!, 10)
-    if (start > 0 && start <= end && end <= 65535) {
-      return { start, end }
-    }
+// Initialize the product information for the server, including the extension gallery URL.
+thisWithVSCodeParams._VSCODE_PRODUCT_JSON = {
+  extensionsGallery: {
+    serviceUrl: 'https://open-vsx.org/vscode/gallery',
+    itemUrl: 'https://open-vsx.org/vscode/item',
+    resourceUrlTemplate: 'https://open-vsx.org/vscode/unpkg/{publisher}/{name}/{version}/{path}',
+    controlUrl: '',
+    nlsBaseUrl: '',
+    publisherUrl: ''
   }
-  return undefined
```

Comment:
for the client to be able to load server extensions files, some properties of that object are mandatory:
- `quality` (with the same value as the client - `oss`)
- `commit` should be set to `VSCODE_REF`
- `webEndpointUrlTemplate` which is used to check if the client is a valid URL to add the `Access-Control-Allow-Origin` header, it need to have the same origin as the client, so it'll probably need to be a parameter :shrug: